lib/puppet/provider/service/smf.rb

```diff
@@ -80,7 +80,7 @@ def startcmd
     when :maintenance, :degraded
       [command(:adm), :clear, @resource[:name]]
     else
-      [command(:adm), :enable, "-s", @resource[:name]]
+      [command(:adm), :enable, "-rs", @resource[:name]]
```

Comment:
In some cases it seems like recursively enabling services is just the default - ie with systemd, if a service expresses dependencies those dependencies are also run on start. Do you know if there's any specific reason this isn't just the default with solaris? ie are you aware of any reason why this change would cause unexpected issues/breakage for anyone in solaris land?